### PR TITLE
[ios] Fix all warnings

### LINKED ISSUE #4429: Fix compiler warnings
We have many compiler warnings in our projects. Fixing the ones that are related to our code will:
- improve our code’s correctness
- reduce noise
- encourage us to keep our warnings to a minimum in the future

It mostly refers to Xcode warnings as they are more annoying, but it would also be nice to do something with Android Studio warnings.

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

Fixes https://github.com/expo/expo/issues/4429.

# How

- added `Bolts` to `Podfile` to be able to suppress warnings from this specific pod,
- added setting deployment target for all development pods
- went through every warning and fixed

# Test Plan

The project compiles, Camera works, ImagePicker works ok.

### CODE REVIEW

File: ios/Exponent/Versioned/Core/Api/Standalone/Branch/RNBranch.m [hunk 1]
```diff
@@ -173,7 +173,7 @@ + (void)initSessionWithLaunchOptions:(NSDictionary *)launchOptions isReferrable:
         if (error) result[RNBranchLinkOpenedNotificationErrorKey] = error;
         if (params) {
             result[RNBranchLinkOpenedNotificationParamsKey] = params;
-            BOOL clickedBranchLink = params[@"+clicked_branch_link"];
+          BOOL clickedBranchLink = [params[@"+clicked_branch_link"] boolValue];
```
Comment (tsapeta) [inline ios/Exponent/Versioned/Core/Api/Standalone/Branch/RNBranch.m]:
```suggestion
            BOOL clickedBranchLink = [params[@"+clicked_branch_link"] boolValue];
```
what happened with the indentation? 🤔

**Review by tsapeta — APPROVED** [review]
Looks good 👍 I've also got an idea to make a macro that would throw compiler errors once specific SDK is phased out - we could use such macro instead of these `// TODO` comments.

**Review by bbarthec — APPROVED** [review]
LGTM 💯 Let's follow @tsapeta suggestion and macro all the code! 😄

**Comment by sjchmiela** [comment]
I agree, created an issue https://github.com/expo/expo/issues/4740

**Pull request merged**